mutt/file.c

```diff
@@ -216,17 +216,10 @@ void mutt_file_unlink(const char *s)
     return;
   }
 
-  FILE *fp = fdopen(fd, "r+");
-  if (fp)
+  unlink(s);
+  if (fd)
   {
-    unlink(s);
-    char buf[2048] = { 0 };
-    while (st.st_size > 0)
-    {
-      fwrite(buf, 1, MIN(sizeof(buf), st.st_size), fp);
-      st.st_size -= MIN(sizeof(buf), st.st_size);
-    }
-    mutt_file_fclose(&fp);
+    mutt_file_fclose(&fd);
```

Comment:
`mutt_file_fclose` takes a `FILE**`, you have an `int fd` :) you can just `close(fd)` it.

Reply:
Right. Corrected. 😊